src/neo/Oracle/Protocols/Https/OracleHttpsProtocol.cs

```diff
@@ -0,0 +1,211 @@
+using Neo.Ledger;
+using Neo.SmartContract.Native;
+using Neo.SmartContract.Native.Oracle;
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Neo.Oracle.Protocols.Https
+{
+    internal class OracleHttpsProtocol
+    {
+        private long _lastHeight = -1;
+
+        /// <summary>
+        /// Config
+        /// </summary>
+        public HttpConfig Config { get; internal set; }
+
+        /// <summary>
+        /// Allow private host
+        /// </summary>
+        public bool AllowPrivateHost { get; internal set; } = false;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public OracleHttpsProtocol()
+        {
+            LoadConfig();
+        }
+
+        /// <summary>
+        /// Load config
+        /// </summary>
+        private void LoadConfig()
+        {
+            // Check if it's the same
+
+            var height = Blockchain.Singleton.Height;
+            if (Interlocked.Exchange(ref _lastHeight, height) == height)
+            {
+                return;
+            }
+
+            // Load the configuration
+
+            using (var snapshot = Blockchain.Singleton.GetSnapshot())
+            {
+                Config = NativeContract.Oracle.GetHttpConfig(snapshot);
+            }
+        }
+
+        // <summary>
+        /// Process HTTP oracle request
+        /// </summary>
+        /// <param name="request">Request</param>
+        /// <returns>Oracle result</returns>
+        public OracleResponse Process(OracleHttpsRequest request)
+        {
+            Log($"Downloading HTTPS request: url={request.URL.ToString()} method={request.Method}", LogLevel.Debug);
+            LoadConfig();
+
+            if (!AllowPrivateHost && IsInternal(Dns.GetHostEntry(request.URL.Host)))
+            {
+                // Don't allow private host in order to prevent SSRF
+
+                LogError(request.URL, "PolicyError");
+                return OracleResponse.CreateError(request.Hash);
+            }
+
+            Task<HttpResponseMessage> result;
+            using var handler = new HttpClientHandler
+            {
+                // TODO: Accept all certificates
+                ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator
+            };
+            using var client = new HttpClient(handler);
+
+            client.DefaultRequestHeaders.Add("Accept", string.Join(",", HttpConfig.AllowedFormats));
+
+            switch (request.Method)
+            {
+                case HttpMethod.GET:
+                    {
+                        result = client.GetAsync(request.URL);
```

Comment:
I have a good idea. We can add a timestamp or block height to the URL when the actual request is made, so that non-determinacy can be eliminated, and the URL has cross-block availability. After solving this problem, we seem to be able to use the synchronization solution.

The user or contract can choose to add a time stamp URL parameter, or block height. Everything is specified in OracleRequest.

Reply:
The conflicts was resolved, but `OracleContract` require a new review.

Reply:
@shargon Do you think that the oracle nodes can reach a consensus on the request timestamp of a transaction? Maybe a node is needed to write the unified timestamp into the contract storage first?

Reply:
> Do you think that the oracle nodes can reach a consensus on the request timestamp of a transaction?

I think that is possible, sometimes, but if don't, we have two possibilities: the first one, it doesn't care because the response was the same, and the worst one, it was different, currently they return a `FAULT` but we can change this to have more retries.